src/liger_kernel/transformers/model/falcon_h1.py

```diff
@@ -85,24 +86,32 @@ def lce_forward(
         # By default, if in training mode, don't materialize logits
         skip_logits = self.training and labels is not None
 
+    # Compute loss
     if skip_logits:
-        loss = LigerForCausalLMLoss(
+        result = LigerForCausalLMLoss(
             hidden_states=kept_hidden_states,
             lm_head_weight=self.lm_head.weight,
             labels=labels,
             shift_labels=shift_labels,
             hidden_size=self.config.hidden_size,
             **kwargs,
         )
+        # Unpack loss and token_accuracy if returned as tuple
+        if isinstance(result, tuple):
+            loss, token_accuracy = result
+        else:
+            loss = result
```

Comment:
The code before your PR assumed that `return_z_loss` was not passed within the `kwargs`: they assumed only the loss scalar was returned.
- Do you know if this assumption is justified?
- If not, then the returned type could be a 2 or 3-tuple (besides the scalar loss), with z_loss and/or token_accuracy